Sources/SwiftDependencyGraphs/DependencyGraph.swift

```diff
@@ -39,4 +39,39 @@ public struct DependencyGraph<V> where V: Hashable, V: Identifiable {
     func contains(vertexWith predicate: (V) -> Bool) -> Bool {
         return vertices.contains(where: {_, vertex in predicate(vertex)})
     }
+    
+    /**
+     Traverses the vertices in the dependency graph and reduces the visited vertices.
+     */
+    public func dfs<T>(
+        startingFrom vertex: V,
+        in direction: TraverseDirection,
+        withVisited visited: Set<V>,
+        reduceWith reducer: (_ accumulator: T, _ currentVertex: V) -> T,
+        withInitialValue accumulator: T
+    ) -> T {
+        guard let neighbours = direction == TraverseDirection.forwards
+                ? outgoing_edges[vertex.id]
+                : incoming_edges[vertex.id]
```

Comment:
Fine as-is, but here's a suggestion that IMHO turns Swift convention usage to 11 :)

```suggestion
        guard let neighbours = self.neighbors(of: vertex, in: direction) else
```

With an implementation like this:

```swift
func neighbors(
    of vertex: V, 
    in direction: TraverseDirection
) -> OrderedSet<V>? {
    let edges = if direction == .forwards {
        outgoing_edges
    } else {
        incoming_edges
    }
    return edges[vertex.id]
}
```

None of that is necessary; I'm exaggerating a bit to show potential Swift idioms.